CMake error if project use `find_package(alpaka)` and does not specifiy a project version

# Requirements

Install alpaka via `cmake --install`: https://alpaka3.readthedocs.io/en/latest/basic/install.html#install-alpaka

# Reproducer

Define a `CMakeLists.txt` which defines a project without version number and import alpaka via `find_package(alpaka)`.

CMakeLists.txt
```cmake
cmake_minimum_required(VERSION 3.19)

project(alpaka_project LANGUAGES CXX)

find_package(alpaka REQUIRED)

add_executable(${CMAKE_PROJECT_NAME})
target_sources(${CMAKE_PROJECT_NAME}
    PRIVATE
    main.cpp
)
set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
    CXX_STANDARD 20
)
```

main.cpp
```cpp
int main(int argc, char **argv){
   return 0;
}
```

If I run CMake configure, I get the following CMake error:

```bash
$ cmake ..
-- The CXX compiler identification is GNU 15.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /home/simeon/.local/micromamba/envs/alpaka3-install-bug/lib/cmake/alpaka/alpakaCommon.cmake:117 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
Call Stack (most recent call first):
  /home/simeon/.local/micromamba/envs/alpaka3-install-bug/lib/cmake/alpaka/alpakaConfig.cmake:46 (include)
  CMakeLists.txt:5 (find_package)


-- Alpaka version: version-NOTFOUND
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5") found components: CXX
-- OpenMP found: 4.5
-- std::atomic_ref<T> found
-- Configuring incomplete, errors occurred!
```

The reason is, because this line sets an empty version variable: https://github.com/alpaka-group/alpaka3/blob/dev/cmake/alpakaCommon.cmake#L117

# Workaround

Set a version number in the project, which includes alpaka.
```cmake
cmake_minimum_required(VERSION 3.19)

project(alpaka_project VERSION 42.0 LANGUAGES CXX)

find_package(alpaka REQUIRED)

add_executable(${CMAKE_PROJECT_NAME})
target_sources(${CMAKE_PROJECT_NAME}
    PRIVATE
    main.cpp
)
set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
    CXX_STANDARD 20
)
``` 3.19)

project(alpaka_project LANGUAGES CXX)

find_package(alpaka REQUIRED)

add_executable(${CMAKE_PROJECT_NAME})
target_sources(${CMAKE_PROJECT_NAME}
    PRIVATE
    main.cpp
)
set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES
    CXX_STANDARD 20
)
```

This solves the problem, but the CMake is still wrong. It displays `42.0` as version number of alpaka.

```bash
$ cmake ..
-- The CXX compiler identification is GNU 15.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Alpaka version: 42.0
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5") found components: CXX
-- OpenMP found: 4.5
-- std::atomic_ref<T> found
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /home/simeon/programming/cpp/alpaka_project/build
```

# Possible Solution

[Line 117](https://github.com/alpaka-group/alpaka3/blob/dev/cmake/alpakaCommon.cmake#L117) should not use the variable `${PROJECT_VERSION}` because the version depends on the project who is using alpaka. Instead we should use the variables `"${alpaka_VERSION_MAJOR}.${alpaka_VERSION_MINOR}.${alpaka_VERSION_PATCH}"` which we extract from the header file.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,6 +97,8 @@ include(GNUInstallDirs)
 
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
+write_basic_package_version_file("alpakaConfigVersion.cmake" VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
+
 install(
     FILES
         "${_alpaka_ROOT_DIR}/cmake/finalize.cmake"
@@ -106,6 +108,7 @@ install(
         "${_alpaka_ROOT_DIR}/cmake/alpakaOneApi.cmake"
         "${_alpaka_ROOT_DIR}/cmake/buildDependencies.cmake"
         "${_alpaka_ROOT_DIR}/cmake/common.cmake"
+        "${PROJECT_BINARY_DIR}/alpakaConfigVersion.cmake"
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/alpaka
 )
 

--- a/cmake/alpakaCommon.cmake
+++ b/cmake/alpakaCommon.cmake
@@ -114,7 +114,7 @@ if(NOT TARGET alpaka)
     target_compile_definitions(alpaka_target_headers INTERFACE ALPAKA_CMAKE_TARGET_HEADERS)
 
     add_library(alpaka INTERFACE)
-    set_target_properties(alpaka PROPERTIES VERSION ${PROJECT_VERSION})
+    set_target_properties(alpaka PROPERTIES VERSION ${alpaka_VERSION})
 
     get_target_property(version alpaka VERSION)
     message(STATUS "Alpaka version: ${version}")

